README.md

```diff
@@ -6,14 +6,44 @@ This is a fork of the excellent [`fuser`](https://github.com/cberner/fuser) Rust
+5. We might want to create a "read-only" PR from the rebased branch with `awslabs:fuser/fork` as the base branch. This PR should never get merged as it'll produce a new commit, but this is a good way for us to have a collaborative consensus on what we're pulling in. After the PR is reviewed it needs to be closed
+> *Note* If we're sure what we're doing the PR step can be skipped and we can force push rebased branch into `fuser/fork`
+6. After PR is approved or changes pushed directly to `awslabs:fuser/fork` we need to create a tag for persisting the branch history.
+    1. Create a new tag with descriptive name containing the currently used fork version, e.g `git tag fuser-fork-0.2.0`.
+    2. Push the created tag to remote `git push upstream fuser-fork-0.2.0`.
+    3. Make sure it's there `git ls-remote --tags upstream "refs/tags/fuser-fork-0.2.0"`.
+It should produce something like `bc31e4d2c2f4601d51b9f0ec37159ee515280253 refs/tags/fuser-fork-0.2.0`
+7. Once `fuser/fork` is in the desired state and the tag is ready we can consume the change in the `main` branch
+    1. `git checkout main`
+    2. Pull in the changes from upstream including submodules `git pull upstream main --recurse-submodules` and update the submodule `git submodule update --remote mountpoint-s3-fuser`.
+> We can avoid adding `--recurse-submodules` argument all the time by telling git to update submodules on every pull `git config set submodule.recurse true`
+
+    1. Since we changed the `fuser/fork` branch and `mountpoint-s3-fuser` refers to it submodule update should produce changes in `mountpoint-s3-fuser` index as well as in `Cargo.lock` if fork's version was changed. We need to commit these changes
+    2. Make sure that everything works by running tests and/or other checks and create a PR to `main` branch
+    3. Get the PR merged
+
+
+#### PR to `fuser/fork`
+
+Create a new branch from `fuser/fork`, commit changes and create a PR with `awslabs:fuser/fork` as the base branch.
+
+The checks for this PR will be broken as they run against the `main` branch which has different structure (this is a point for improvement), but we still can have a consensus on what we're pulling in.
+After PR is reviewed it needs to be closed (as will don't want to further modify the history via GitHub Merge) and the commit needs to be cherry-picked on top of `fuser/fork` branch `git cherry-pick -m 1 <commit-hash>`
+
+After that we can push the updated local `fuser/fork` branch to remote.
+
+#### Publishing the fork
+We want to treat the fork as a standard crate inside our project and follow the existing [manual](https://github.com/awslabs/mountpoint-s3/blob/main/mountpoint-s3-client/PUBLISHING_CRATES.md) for publishing.
```

Comment:
Can we really follow that guide as it is? I'm wondering what we're meant to do in the post-release steps, where we go and update the version. Should we go and do that?

Reply:
I think rebasing should produce a new version of `mountpoint-s3-fuser`. We need to consume it by updating the [dependency](https://github.com/awslabs/mountpoint-s3/blob/main/mountpoint-s3/Cargo.toml#L10C1-L10C89) in MP, e.g.:
```
fuser = { path = "../mountpoint-s3-fuser", version = "0.1.0", features = ["abi-7-28"] }
```
gets updated to 
```
fuser = { path = "../mountpoint-s3-fuser", version = "0.2.0", features = ["abi-7-28"] }
```

This should be done via PR to `awslabs/mounpoint-s3/main`, in the same PR which updates the version of the submodule. This is where the MP tests are run to verify the rebase result. 

I think we don't need PRs to  `awslabs/mounpoint-s3/fuser/fork` and can just force push to it, relying on the PR to `main` to catch regressions.

Reply:
Ah, never mind, I guess the testing bit is already covered by [the previous step](https://github.com/awslabs/mountpoint-s3/pull/1283/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R32). My point is that checks on the PR to the main branch make it ok to force push to `fuser/fork`.